pyannote/database/filelist.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# The MIT License (MIT)
+
+# Copyright (c) 2016 CNRS
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Hervé BREDIN - http://herve.niderb.fr
+
+from pyannote.core import Segment, Timeline, Annotation
+from .database import Database
+from . import protocol as Protocol
+import functools
+from pathlib import Path
+import yaml
+import pandas as pd
+
+
+def subset_iter(database_name, path_mdtm=None, path_uem=None):
+    """This function will become a xxx_iter method of a protocol.
+
+    Parameters
+    ----------
+    database_name : `str`
+        Database name.
+    path_mdtm : `Path`, optional
+        Path to MDTM file.
+    path_uem : `Path`, optional
+        Path to UEM file.
+
+    Yields
+    ------
+    current_file : `dict`
+        Dictionary that provides information about a file.
+        This includes (at least) a 'uri' key and a 'database' keys.
+
+    """
+
```

Comment:
Start by gathering uris from `path_lst` and then only iterate over the corresponding annotation groups and annotated groups (if they exist).